packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart

```diff
@@ -96,9 +96,9 @@ mixin ShapeHitbox on ShapeComponent implements Hitbox<ShapeHitbox> {
 
     // This should be placed after the hitbox parent listener
     // since the correct hitbox size is required by the QuadTree.
-    final parentGame = findParent<FlameGame>();
-    if (parentGame is HasCollisionDetection) {
-      _collisionDetection = parentGame.collisionDetection;
+    final parent = findParent<HasCollisionDetection>();
```

Comment:
maybe note that this will find the "closest" parent that has the mixin (and that that is intended)

Reply:
Good idea, I haven't updated the docs for this PR yet, that should be done before it is merged.

Reply:
Added docs and an example too!